src/Zend/Uri/Http.php

```diff
@@ -191,11 +191,6 @@ protected function _parseUri($schemeSpecific)
             throw new Zend_Uri_Exception('Internal error: scheme-specific decomposition failed');
         }
 
-        // Failed decomposition; no further processing needed
-        if ($status === false) {
```

Comment:
This same check is done a few lines above, and an exception is thrown if `$status` is `false`, so this second check isn't necessary.